scheduling/taskmetadata.go

```diff
@@ -28,18 +32,48 @@ type TaskMetadataManager struct {
 	idGenerator ffsync.OrderedIdGenerator
 }
 
-func NewMemoryTaskMetadataManager() TaskMetadataManager {
-	memoryLocker := ffsync.NewMemoryLocker()
-	memoryStorage := ss.NewMemoryStorageImplementation()
+func NewMemoryTaskMetadataManager() (TaskMetadataManager, error) {
```

Comment:
with the added error return, you'll need to update `metadata.go:1482`

`taskManager := scheduling.NewMemoryTaskMetadataManager()` (needs error)